### DISCUSSION TITLE
Strange behavior of MyTourbook

### DISCUSSION THREAD

**Discussion by ovahead** [mytourbook/mytourbook#1057] [Q&A]
This affects the Tour List, Tour Graphic and 2D Tour Map view
When saving a tour from the current month/day that has already been saved, the current month and tour will be closed (Tour List view) and a tour from the previous month is opened in all views mentioned above. And every time the tour the 27.03.23 will be open.
Only the Tour Editor still shows the saved tour

Is there possibly any option to adjust this in MTB?
I am using MyTourbook 23.3. And I really dont know if this behavior was aleady in 22.11

edit:
on tours in march and februar the views jumped also the tour from 27.03.23

**Comment by ovahead** [comment]
It was a one-time behavoir and I can not repeat it. Maybe a missing memory error or what ever.

**Comment by FJBDev** [comment]
When/if you repeat it, can you take screenshots as well ?
I am not even sure I fully understand the issue you encountered and a picture is usually worth 1000 words.

Cheers

**Reply by ovahead** [reply]
If this will happen again I think about a screenshot.  If possiblle a privat mail because not too much persons should see my data ;)

**Reply by wolfgang-ch** [reply]
With `-DscrambleData` in the command line https://github.com/mytourbook/mytourbook#user-content-other, some parts in MT can be scramled

**Comment by wolfgang-ch** [comment]
A possible problem could be that the selection provider from opened or hidden views (or hidden perspectives) can fire a selection. 2 days ago I fixed such an issue.

Selections in MT can be very complex and time consuming to debug and fix it.

https://github.com/mytourbook/mytourbook/blob/5b38af6ba90a5bfd82c67b8c51c65dd157bbec5f/bundles/net.tourbook/src/net/tourbook/ui/views/tourBook/TourBookView.java#L3365-L3369

**Comment by ovahead** [comment]
Okay now I have an other issue with MyTourbook.
Everytime I start MT the 2D Tour Map view starts with SRTM as map provider after changing it once to SRTM. If I change the map provider to OSM and click apply and close, next time MT starts again with SRTM as map provider :(

I know that OSM had an green circle before I tried to change it to SRTM but now both provider had always a red circle before the name of the provider.

**Reply by FJBDev** [reply]
@ovahead Can you share screenshots of your issues?

**Reply by wolfgang-ch** [reply]
@ovahead Can you create a new issue item for this "...other issue"
